### ISSUE TITLE
[Java] Implement new concept exercise: booleans

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#1779]
This issue describes how to implement the `booleans` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise][anatomy-of-a-concept-exercise].

## Goal

The goal of this exercise is to learn about boolean operation

## Learning objectives

- Know of the existence of the `boolean` type and its two values.
- Know about boolean operators and how to build logical expressions with them.
- Know of the boolean operator precedence rules.

## Out of scope

Nothing to report

## Concepts

- `booleans`: know of the existence of the `boolean` type and its two values; know about boolean operators and how to build logical expressions with them; know of the boolean operator precedence rules.

## Prequisites

This exercise's prerequisites Concepts are:

- `basics`: know how to define methods.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer)

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer)

## Implementing
To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by samuelteixeiras** [comment]
Except for the video link, the links below are going to a 404 page:

- The features of v3.

- Rationale for v3.

- What are concept exercise and how they are structured?

**Comment by mirkoperillo** [comment]
Hi @samuelteixeiras thank you to signal it. Fixed.

Feel free to ask to every question or doubt
